core/services/postgres/event_broadcaster.go

```diff
@@ -0,0 +1,326 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"sync"
+	"time"
+
+	"github.com/lib/pq"
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+//go:generate mockery --name EventBroadcaster --output ./mocks/ --case=underscore
+//go:generate mockery --name Subscription --output ./mocks/ --case=underscore
+
+// EventBroadcaster opaquely manages a collection of Postgres event listeners
+// and broadcasts events to subscribers (with an optional payload filter).
+type EventBroadcaster interface {
+	Start() error
+	Stop() error
+	Subscribe(channel, payloadFilter string) (Subscription, error)
+	Notify(channel string, payload interface{}) error
+}
+
+type eventBroadcaster struct {
+	uri                  string
+	minReconnectInterval time.Duration
+	maxReconnectDuration time.Duration
+	db                   *sql.DB
+
+	listeners   map[string]*channelListener
+	listenersMu sync.Mutex
```

Comment:
We are inconsistent whether we use a pointer or embed the mutex directly on the struct. Does it matter?

Reply:
It's probably best to not embed (anonymously) so that nothing external can try to `.Lock()` it.  Regarding pointer vs. non-pointer, given that it's a private field and nothing else can see it/take a reference to it, it really doesn't make a difference.  I like using pre-initialized zero values whenever possible, which `sync.Mutex` was designed for.

Reply:
I suppose I mean if you copy the struct, wouldn't it copy the mutex too?

Reply:
`sync.Mutex` has special handling -- it's disallowed from being copied.  I believe `go vet` is what surfaces the error when you try to do so